packages/bot/src/stores/quick-strategy-store.js

```diff
@@ -98,41 +110,128 @@ export default class QuickStrategyStore {
         this.toggleStrategyModal();
     }
 
+    validateQuickStrategy = values => {
+        const errors = {};
+        const number_field = ['duration', 'stake', 'size', 'profit', 'loss'];
+    
+        Object.keys(values).forEach(key => {
+            const value = values[key];
+    
+            if (number_field.includes(key)){
+                if (isNaN(value)) {
+                    errors[key] = translate('Must be a number');
+                }
+    
+                if (value <= 0){
+                    errors[key] = translate('Must be a number higher than 0');
+                }
+    
+                if (/^0+(?=\d)/.test(value)) {
+                    errors[key] = translate('Invalid number format');
+                }
+            }
+    
+            if (value === '') {
+                errors[key] = translate('Field cannot be empty');
+            }
+        });
+
+        const { min, max } = this.duration_dropdown.filter(duration => duration.unit === values.duration_type)[0];
+        if (values.duration < min) {
+            errors.duration = `${translate('Minimum duration :')} ${min}`;
+        }
+
+        if (values.duration > max) {
+            errors.duration = `${translate('Maximum duration :')} ${max}`;
+        }
+    
+        return errors;
+    };
+
     @action.bound
     async onMount() {
         const { active_symbols } = ApiHelper.instance;
         const market_options     = await active_symbols.getAssetOptions();
 
-        this.updateTradetypeDropdown();
+        await this.updateTradetypeDropdown();
+        await this.updateDurationDropdown();
+        await this.updateDurationValue();
             
         runInAction(() => {
             this.market_dropdown = market_options;
         });
     }
 
     @action.bound
-    async updateTradetypeDropdown(symbol = this.initial_values.symbol) {
+    // eslint-disable-next-line
+    async onChangeMarketDropdown(setFieldValue, value) {
+        setFieldValue('symbol', value);
+        await this.updateTradetypeDropdown(setFieldValue, value);
+
+        const first_trade_type_option = this.trade_type_dropdown[Object.keys(this.trade_type_dropdown)[0]][0].value;
+
+        setFieldValue('trade_type', first_trade_type_option);
+    }
+
+    @action.bound
+    async updateTradetypeDropdown(setFieldValue, symbol = this.initial_values.symbol) {
```

Comment:
Since `setFieldValue` does not always have a value it would be good to mark this as an optional argument by supplying a default value of e.g. `null`.

```suggestion
    async updateTradetypeDropdown(setFieldValue = null, symbol = this.initial_values.symbol) {
```